src/FetchLoader.js

```diff
@@ -46,6 +16,23 @@ class FetchLoader extends React.Component {
     }
   }
 
+  static getDerivedStateFromProps(props, state) {
+    const url = URI(props.resource, props.host).toString()
+    // Store url in state so we can compare when props change.
+    // Clear out previously-loaded data (so we don't render stale stuff).
+    if (url !== state.url) {
+      return {
+        data: null,
+        loading: true,
+        error: null,
+        url: url
```

Comment:
I am totally unfamiliar with this state-of-the-art method, but it seems an improved lifecycle method in React. Personally, it will be easier for me to understand if we use `nextprops` and `prevstate` as the inputs. Besides, should we set an initial `url` in the component's `state`?

Reply:
`nextProps` and `nextState` are part of `componentWillUpdate` and `compononetWillReceiveProps`. Those two methods together with `componentWillMount` are going awaya and [the new lifecycle ](https://reactjs.org/docs/react-component.html#the-component-lifecycle) uses `getDerivedStateFromProps`, which is what we’ll be using from now on.

In [the React guide on async rendering](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#fetching-external-data) tge `prevId` fuekd isn’t part of the initial state so I think it’s safe to have it added in `getDerivedStateFromProps` instead.